### PR TITLE
feat: mls epic

### PR DESCRIPTION
**Pull request by typfel**
----
#### PR Submission Checklist for internal contributors

- The **PR Title**
    - [ ] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
    - [ ] contains a reference JIRA issue number like `SQPIT-764`
    - [ ] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
    - [ ] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

Build from MLS epic for migration and 1-1 support.

- https://github.com/wireapp/wire-android-reloaded/pull/2128
- https://github.com/wireapp/wire-android-reloaded/pull/2277
- https://github.com/wireapp/wire-android-reloaded/pull/2327
- 
----
#### PR Post Submission Checklist for internal contributors (Optional)

- [ ] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

- [ ] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

**Comment by AndroidBob** [comment]
Build [463](https://10.10.124.134/job/AR-build-pipeline/463/) **failed**.

**Comment by AndroidBob** [comment]
Build [474](https://10.10.124.134/job/AR-build-pipeline/474/) **failed**.

File: app/src/main/kotlin/com/wire/android/ui/home/conversations/model/UIMessage.kt [hunk 1]
```diff
@@ -367,13 +368,34 @@ sealed class UIMessageContent {
...
+                Conversation.Protocol.MLS -> R.string.label_system_message_conversation_protocol_changed_mls
+            }
+        )
+
+        class HistoryLost: SystemMessage(
+            R.drawable.ic_info,
+            R.string.label_system_message_conversation_history_lost,
+            true)
+
+        class HistoryLostProtocolChanged: SystemMessage(
```
Comment (borichellow) [inline app/src/main/kotlin/com/wire/android/ui/home/conversations/model/UIMessage.kt]: `HistoryLostProtocolChanged` and `HistoryLost` could be just `object`  not `class`

File: app/src/main/res/values/strings.xml [hunk 2]
```diff
@@ -567,6 +567,10 @@
     <string name="label_system_message_deactivated">deactivated</string>
     <string name="label_system_message_conversation_history_lost">You haven\'t used this device for a while. some messages may not appear here.</string>
     <string name="label_system_message_conversation_mls_wrong_epoch_error_handled">The MLS group key was updated without our knowledge. This could happen due to lost messages between backends, or a bug. We have automatically rejoined the conversation, but you may have lost messages.</string>
+    <string name="label_system_message_conversation_history_lost_protocol_changed">You didn\'t update your app in time. some messages may not appear here.</string>
+    <string name="label_system_message_conversation_protocol_changed_proteus">Migration of encryption protocol was canceled.</string>
+    <string name="label_system_message_conversation_protocol_changed_mixed">Migration of encryption protocol has started. Make sure you all your Wire clients are updated.</string>
```
Comment (borichellow) [inline app/src/main/res/values/strings.xml:598]: `Migration of encryption protocol has started. Make sure all your Wire clients are updated.`

File: app/src/main/res/values/strings.xml [hunk 3]
```diff
@@ -567,6 +567,10 @@
     <string name="label_system_message_deactivated">deactivated</string>
     <string name="label_system_message_conversation_history_lost">You haven\'t used this device for a while. some messages may not appear here.</string>
     <string name="label_system_message_conversation_mls_wrong_epoch_error_handled">The MLS group key was updated without our knowledge. This could happen due to lost messages between backends, or a bug. We have automatically rejoined the conversation, but you may have lost messages.</string>
+    <string name="label_system_message_conversation_history_lost_protocol_changed">You didn\'t update your app in time. some messages may not appear here.</string>
```
Comment (borichellow) [inline app/src/main/res/values/strings.xml]: `You didn\'t update your app in time. Some messages may not appear here.`

**Review by borichellow — APPROVED** [review]
Nice one! 
Just a small comments

File: app/src/main/kotlin/com/wire/android/ui/home/conversations/model/UIMessage.kt (hunk 1 above)
Comment (typfel) [reply app/src/main/kotlin/com/wire/android/ui/home/conversations/model/UIMessage.kt]: Thanks for the review! I'm mainly trying to produce a build for Antje here, I'll open real PR later when this is ready to be merged.

**Comment by AndroidBob** [comment]
Build [556](https://10.10.124.134/job/AR-build-pipeline/556/) **failed**.

File: app/src/main/res/values/strings.xml (hunk 3 above)
Comment (typfel) [reply app/src/main/res/values/strings.xml]: Capitalised `Some`.

**Comment by AndroidBob** [comment]
Build [564](https://10.10.124.134/job/AR-build-pipeline/564/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-v4.3.0(7088341)(7088341)-staging-compat.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/staging/compat/com.wire.android-v4.3.0(7088341)(7088341)-staging-compat.apk)

**Comment by AndroidBob** [comment]
Build [572](https://10.10.124.134/job/AR-build-pipeline/572/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-v4.3.0(7089214)(7089214)-staging-compat.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/staging/compat/com.wire.android-v4.3.0(7089214)(7089214)-staging-compat.apk)

**Review by saleniuk — APPROVED** [review]

**Comment by AndroidBob** [comment]
Build [1113](https://10.10.124.134/job/AR-build-pipeline/1113/) **failed**.

**Comment by AndroidBob** [comment]
Build [1114](https://10.10.124.134/job/AR-build-pipeline/1114/) **failed**.

**Comment by AndroidBob** [comment]
Build [1123](https://10.10.124.134/job/AR-build-pipeline/1123/) **failed**.

**Comment by AndroidBob** [comment]
Build [1133](https://10.10.124.134/job/AR-build-pipeline/1133/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-v4.4.0(7322222)(7322222)-staging-compat.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/staging/compat/com.wire.android-v4.4.0(7322222)(7322222)-staging-compat.apk)

**Comment by AndroidBob** [comment]
Build [1265](https://10.10.124.134/job/AR-build-pipeline/1265/) **failed**.

**Comment by AndroidBob** [comment]
Build [572](https://10.10.124.134/job/AR-build-pipeline/572/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-v4.5.0(7495106)(7495106)-staging-compat.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/staging/compat/com.wire.android-v4.5.0(7495106)(7495106)-staging-compat.apk)

**Comment by AndroidBob** [comment]
Build [673](https://10.10.124.134/job/AR-build-pipeline/673/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-v4.5.0(7521072)(7521072)-staging-compat.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/staging/compat/com.wire.android-v4.5.0(7521072)(7521072)-staging-compat.apk)

**Comment by AndroidBob** [comment]
Build [1010](https://10.10.124.134/job/AR-build-pipeline/1010/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-v4.5.0(7679810)(7679810)-staging-compat.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/staging/compat/com.wire.android-v4.5.0(7679810)(7679810)-staging-compat.apk)

**Comment by AndroidBob** [comment]
Build [1037](https://10.10.124.134/job/AR-build-pipeline/1037/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-v4.5.0(7687689)(7687689)-staging-compat.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/staging/compat/com.wire.android-v4.5.0(7687689)(7687689)-staging-compat.apk)

**Comment by AndroidBob** [comment]
Build [1178](https://10.10.124.134/job/AR-build-pipeline/1178/) **failed**.

**Comment by AndroidBob** [comment]
Build [1179](https://10.10.124.134/job/AR-build-pipeline/1179/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-v4.5.0(7762477)(7762477)-staging-compat.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/staging/compat/com.wire.android-v4.5.0(7762477)(7762477)-staging-compat.apk)

**Comment by AndroidBob** [comment]
Build [1215](https://10.10.124.134/job/AR-build-pipeline/1215/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-v4.5.0(7796291)(7796291)-staging-compat.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/staging/compat/com.wire.android-v4.5.0(7796291)(7796291)-staging-compat.apk)

**Comment by AndroidBob** [comment]
Build [1225](https://10.10.124.134/job/AR-build-pipeline/1225/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-v4.5.0(7798310)(7798310)-staging-compat.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/staging/compat/com.wire.android-v4.5.0(7798310)(7798310)-staging-compat.apk)

**Comment by AndroidBob** [comment]
Build [1229](https://10.10.124.134/job/AR-build-pipeline/1229/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-v4.5.0(7805049)(7805049)-staging-compat.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/staging/compat/com.wire.android-v4.5.0(7805049)(7805049)-staging-compat.apk)

**Comment by AndroidBob** [comment]
Build [1233](https://10.10.124.134/job/AR-build-pipeline/1233/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-v4.5.0(7805444)(7805444)-staging-compat.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/staging/compat/com.wire.android-v4.5.0(7805444)(7805444)-staging-compat.apk)

**Comment by AndroidBob** [comment]
Build [1240](https://10.10.124.134/job/AR-build-pipeline/1240/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-v4.5.0(7806183)(7806183)-staging-compat.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/staging/compat/com.wire.android-v4.5.0(7806183)(7806183)-staging-compat.apk)

**Comment by AndroidBob** [comment]
Build [1248](https://10.10.124.134/job/AR-build-pipeline/1248/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-v4.5.0(7809271)(7809271)-staging-compat.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/staging/compat/com.wire.android-v4.5.0(7809271)(7809271)-staging-compat.apk)

**Comment by AndroidBob** [comment]
Build [1283](https://10.10.124.134/job/AR-build-pipeline/1283/) **failed**.

**Comment by AndroidBob** [comment]
Build [1284](https://10.10.124.134/job/AR-build-pipeline/1284/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-v4.5.0(7821001)(7821001)-staging-compat.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/staging/compat/com.wire.android-v4.5.0(7821001)(7821001)-staging-compat.apk)

**Comment by AndroidBob** [comment]
Build [1307](https://10.10.124.134/job/AR-build-pipeline/1307/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-v4.5.0(7844174)(7844174)-staging-compat.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/staging/compat/com.wire.android-v4.5.0(7844174)(7844174)-staging-compat.apk)

**Comment by AndroidBob** [comment]
Build [1308](https://10.10.124.134/job/AR-build-pipeline/1308/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-v4.5.0(7844489)(7844489)-staging-compat.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/staging/compat/com.wire.android-v4.5.0(7844489)(7844489)-staging-compat.apk)

**Pull request merged**